app/src/main/java/uk/ac/bath/petmatch/ShelterEditProfileActivity.java

```diff
@@ -32,9 +32,26 @@ protected void onCreate(Bundle savedInstanceState) {
 
         User currentUser = loginService.getLoggedInUser();
         ShelterDao shelterDao = db.shelters;
-        currentShelter = shelterDao.findByTitle("Dummy shelter");
 
-        if((currentUser == null) || !(currentUser.getShelter().toString().equals(currentShelter.toString()))) {
+        if(currentUser == null) {
+
+            showPermissionError();
+            return;
+        }
+
+        if(currentUser.getShelter() == null) {
+
+            showPermissionError();
+            return;
+        }
+        currentShelter = shelterDao.findById(currentUser.getShelter().getId());
```

Comment:
The permission checking is good

I like that you reload the current shelter from DB every time, instead of using the possibly stale version in user.getShelter()